### PR TITLE
Do not cache share links

### PR DESCRIPTION
**Pull request by melroy89**
You can't just cache the share links, that will result into every blog post having the same URL.

Just use `partial` on the getShareLink.html. So now each post will show the correct URL to share the post via social media or alike.

### CODE REVIEW

File: layouts/partials/social/share.html [hunk 1]
```diff
@@ -34,7 +34,7 @@
         {{/* @todo notification into CLI that a network is configured but not supported */}}
       {{- end -}}
       {{- $options := (dict "context" page "setup" $setup) }}
-      {{- $href := partialCached "func/getShareLink.html" $options $options -}}
+      {{- $href := partial "func/getShareLink.html" $options $options -}}
```
Comment (melroy89) [inline layouts/partials/social/share.html:37]: Why is there two times `$options`?

Comment (melroy89) [reply layouts/partials/social/share.html:37]: Since we now use `partial` instead of `partialCached`, I believe we can just pass 1x the `$options`.

**Comment by davidsneighbour** [comment]
False. There is a bug in the logic of the partial, that is true. But we CAN and will cache those links. A share link must be cacheable for a specific page. Check the $options dict. `page` here might refer to the outer list page a post is listed on, which is what ruins the cache if you implement share links on list pages. What this needs is `$context` (defined in the first line) instead of `page` in the options and then they (should be/are) individual caches per post object (if `$context` doesn't walk into the same trap). I'll need to check that though, it might be that I will inline the sharelink generation.

File: layouts/partials/social/share.html [hunk 2]
```diff
@@ -34,7 +34,7 @@
         {{/* @todo notification into CLI that a network is configured but not supported */}}
       {{- end -}}
       {{- $options := (dict "context" page "setup" $setup) }}
-      {{- $href := partialCached "func/getShareLink.html" $options $options -}}
+      {{- $href := partial "func/getShareLink.html" $options -}}
```
Comment (davidsneighbour) [inline layouts/partials/social/share.html:37]: nope ;) see my comment. `{{- $options := (dict "context" $context "setup" $setup -}}` should do the trick with unchanged `partialCached` line.

**Review by davidsneighbour — CHANGES_REQUESTED** [review]

File: layouts/partials/social/share.html (hunk 2 above)
Comment (melroy89) [reply layouts/partials/social/share.html:37]: Sorry what? I don't see your comment.

**Comment by davidsneighbour** [comment]
Something along the lines of this, not tested yet:

https://github.com/theNewDynamic/gohugo-theme-ananke/pull/763

**Review by davidsneighbour — APPROVED** [review]

**Comment by davidsneighbour** [comment]
I am going to merge this now. I opened an issue for going to look at caching after the release.

https://github.com/theNewDynamic/gohugo-theme-ananke/issues/764

**Pull request merged**